flask_security/datastore.py

```diff
@@ -255,6 +255,35 @@ def find_role(self, role):
         return self.role_model.query.filter_by(name=role).first()
 
 
+class SQLAlchemySessionUserDatastore(SQLAlchemyUserDatastore,
+                                     SQLAlchemyDatastore):
+    """A SQLAlchemy datastore implementation for Flask-Security that assumes the
+    use of the flask_sqlalchemy_session extension.
+    """
+    def __init__(self, session, user_model, role_model):
+
+        class PretendFlaskSQLAlchemyDb(object):
+            """ This is a pretend db object, so we can just pass in a session.
+            """
+            def __init__(self, session):
+                self.session = session
+
+        SQLAlchemyUserDatastore.__init__(self,
+                                         PretendFlaskSQLAlchemyDb(session),
+                                         user_model,
+                                         role_model)
+
+    def commit(self):
+        # Old flask-sqlalchemy adds this weird attribute for tracking
+        # to Session. flask-sqlalchemy 2.0 does things more nicely.
+        try:
+            self.db.session.commit()
```

Comment:
What about using `super(SQLAlchemySessionUserDatastore, self).commit()` instead on both places in this method?